app/models/docker_registry.rb

```diff
@@ -0,0 +1,53 @@
+# model that represents a docker registry
+class DockerRegistry < ActiveRecord::Base
+  has_one :certificate_service, as: :service, dependent: :destroy
+  has_one :certificate, through: :certificate_service
+
+  validates :url, presence: true, uniqueness: true, url: { schemes: ["https", "http"] }
```

Comment:
@vitoravelino i was not really sure when i added this, should the user also be able to enter `some.url.local` without the protocol prefix? then we'd need to adapt the restriction

Reply:
I dunno... I don't think we have an use case that needs the url without the prefix.

Reply:
In the beginning I would force the user to always enter the url prefix